scripts/templates/print.cpp.mako

```diff
@@ -0,0 +1,105 @@
+<%!
+import re
+from templates import helper as th
+%><%
+    n=namespace
+    N=n.upper()
+
+    x=tags['$x']
+    X=x.upper()
+%>/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file ${n}_print.cpp
+ *
+ */
+
+#include "${n}_print.h"
+#include "${n}_print.hpp"
+
+#include <sstream>
+#include <string.h>
+
+using namespace ${x}_print;
+
+<%def name="ss_copy(item_name)">std::stringstream ss;
+    ss << ${item_name};
+    #if defined(_WIN32)
+    strncpy_s(out, size, ss.str().c_str(), ss.str().size() + 1);
+    #else
+    strncpy(out, ss.str().c_str(), size);
+    #endif
+</%def>
+
+%for spec in specs:
+%for obj in spec['objects']:
+## ENUM #######################################################################
+%if re.match(r"enum", obj['type']):
```

Comment:
I think enums could share implementation with the hpp header.

Reply:
Done. Now functions for printing enums return ur_result_t just like all the other functions from this header.